addons/mail_group/models/mail_group.py

```diff
@@ -0,0 +1,319 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import logging
+from werkzeug import urls
+
+from ast import literal_eval
+
+from odoo import _, api, fields, models, modules
+from odoo.exceptions import ValidationError, UserError
+from odoo.tools import email_normalize, hmac
+
+_logger = logging.getLogger(__name__)
+
+
+class MailGroup(models.Model):
+    """Represent a mailing list.
+
+    The users can send emails to an alias and it will create <mail.group.message>. Those
+    messages can be moderated (accepted, rejected...) or not and all members of the
+    mailing list will receive the email.
+    """
+
+    _name = 'mail.group'
+    _description = 'Mailing List'
+
+    _inherit = ['mail.alias.mixin']
+
+    def _get_default_image(self):
+        image_path = modules.get_module_resource('mail', 'static/src/img', 'groupdefault.png')
+        return base64.b64encode(open(image_path, 'rb').read())
+
+    name = fields.Char('Name', required=True)
+    alias_name = fields.Char('Alias Name', copy=False, related='alias_id.alias_name', readonly=False)
+    description = fields.Text('Description')
+    image_128 = fields.Image('Image', max_width=128, max_height=128, default=_get_default_image)
+    # Messages
+    mail_group_message_ids = fields.One2many('mail.group.message', 'mail_group_id', string='Pending Messages')
+    mail_group_message_count = fields.Integer('Messages Count', help='Messages that need an action', compute='_compute_mail_group_message_count')
```

Comment:
mail_group_message_moderation_count ? As this is the counter of messages waiting moderation if I understood well.